src/ai/backend/agent/agent.py

```diff
@@ -1329,12 +1334,24 @@ async def create_kernel(
             kernel_config['image']['digest'],
             AutoPullBehavior(kernel_config.get('auto_pull', 'digest')),
         )
+
+        # create reporter for kernel-pull-process
+        async def reporter(
+            current: Union[int, float], 
+            total: Union[int, float],
+            ) -> None:
```

Comment:
For multi-line annotated function arguments with trailing return type annotation, dedent before the ending parenthesis like this:
```suggestion
        ) -> None:
```